cmd/juju/application/deploy.go

```diff
@@ -767,6 +774,11 @@ func (c *DeployCommand) deployCharm(
 		appConfig[k] = v.(string)
 	}
 
+	// Expand the trust flag into the appConfig
+	if c.Trust {
+		appConfig[app.TrustConfigOptionName] = strconv.FormatBool(c.Trust)
+	}
```

Comment:
If the flag is present we just splice the value into the configuration parameters. We try to reuse as much of the configuration infrastructure as possible.